components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -327,7 +327,15 @@ public Application updateApplication(ApplicationUpdateRequest updateRequest, Str
             }
             appDTO.setRequestObjectSignatureValidationEnabled(updateRequest.isRequireSignedRequestObject());
             appDTO.setRequirePushedAuthorizationRequests(updateRequest.isRequirePushedAuthorizationRequests());
-            appDTO.setTlsClientCertificateBoundAccessTokens(updateRequest.isTlsClientCertificateBoundAccessTokens());
+            if (updateRequest.isTlsClientCertificateBoundAccessTokens()) {
+                boolean isCertificateTokenBindEnabled = DCRDataHolder.getInstance().getTokenBinders().stream()
+                        .anyMatch(t -> t.getBindingType().equals(
+                                OAuth2Constants.TokenBinderType.CERTIFICATE_BASED_TOKEN_BINDER));
```

Comment:
lets change the check to be null safe.
OAuth2Constants.TokenBinderType.CERTIFICATE_BASED_TOKEN_BINDER.equals(t.getBindingType())

Reply:
Do the change in the other place as well